pkg/lifecycle/lifecycle_test.go

```diff
@@ -0,0 +1,181 @@
+package lifecycle_test
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-psap/special-resource-operator/pkg/clients"
+	"github.com/openshift-psap/special-resource-operator/pkg/lifecycle"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	name      = "test"
+	namespace = "ns"
+)
+
+var (
+	ctrl       *gomock.Controller
+	labels     = map[string]string{"key": "value"}
+	mockClient *clients.MockClientsInterface
+
+	optNs     = client.InNamespace(namespace)
+	optLabels = client.MatchingLabels(labels)
+
+	unstructuredMatcher     = gomock.AssignableToTypeOf(&unstructured.Unstructured{})
+	unstructuredListMatcher = gomock.AssignableToTypeOf(&unstructured.UnstructuredList{})
+)
+
+func TestLifecycle(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	BeforeEach(func() {
```

Comment:
When is this one actually running? Never saw a `BeforeEach` or `AfterEach` outside a `Describe`

Reply:
Just like in a `Describe`, it's running before each of the suite's `It`. @yevgeny-shnaidman came up with this suggestion to have `BeforeEach` when all tests are top-level `Describe`s.